src/Plugin/Field/FieldFormatter/FormatterBase.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Drupal\fb_instant_articles\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase as DrupalFormatterBase;
+
+/**
+ * Base class for all of our FBIA field formatters.
+ */
+class FormatterBase extends DrupalFormatterBase {
+
+  const SOURCE_TYPE_URL = 'url';
+
+  const SOURCE_TYPE_HTML = 'html';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    // Do nothing. Our field formatters are unique in that they are not meant
+    // to generate HTML on their own, but only signal to the Serialization API
+    // the fate of this field in the FBIA document.
```

Comment:
Good question! The way it will work is the Serialization API is given the entity (usually a node) as an argument. It then iterates over the configured fields, each field being an object of type `FieldItemListInterface`. It then calls the field formatter with that object as the first param, so that's how it'll get it. This method signature will change a bit too in https://www.drupal.org/node/2872807.